aws-api/src/main/java/com/amplifyframework/api/aws/AppSyncGraphQLOperation.java

```diff
@@ -159,12 +182,47 @@ public void onFailure(@NonNull Call call, @NonNull IOException exception) {
         }
     }
 
+    static final class MultiAuthInterceptor implements Interceptor {
+        private final Iterator<AuthorizationType> authTypes;
+        private final ApiRequestDecoratorFactory apiRequestDecoratorFactory;
+
+        MultiAuthInterceptor(@NonNull Iterator<AuthorizationType> authTypes,
+                             @NonNull ApiRequestDecoratorFactory apiRequestDecoratorFactory) {
+            this.authTypes = authTypes;
+            this.apiRequestDecoratorFactory = apiRequestDecoratorFactory;
+        }
+
+        @NotNull
+        @Override
+        public Response intercept(@NotNull Chain chain) throws IOException {
+            Request httpRequest = chain.request().newBuilder().build();
```

Comment:
I'm using this as a known starting point, because some of the decorators do modify the request. This basically ensures that any of the decorators always start with the original request.